### ISSUE TITLE
Typescript: Reactive assignments do not infer data type

### ISSUE THREAD

**Issue by dkzlv** [sveltejs/language-tools#271]
**Describe the bug**
Subject.

**To Reproduce**
Have these enabled in `tsconfig.json`:
```
"noImplicitAny": true,
"strictNullChecks": true,
```

and create a component with the following code:
```html
<script lang="typescript">
  export let data: { someKey: string | null };

  $: show = !!data.someKey;
</script>

{#if show}hey{/if}
```
This simple code raises two TS errors:
`Variable 'show' implicitly has type 'any' in some locations where its type cannot be determined.` for the `show` variable in the `script` tag and `Variable 'show' implicitly has an 'any' type` for the same variable in the template.

**Expected behavior**
Seems to be safe enough to infer the data type as boolean.

**System (please complete the following information):**
OS: Mac OS 10.15.3
IDE: VS Code
Plugin/Package: Svelte for VS Code (the new official one)

**Additional context**

Currently the parser does not allow neither this
```typescript
$: show = !!data.someKey as boolean;
```
nor that
```typescript
$: show: boolean = !!data.someKey;
```
The only possible (and quite simple!) solution is to duplicate the variable, which is completely fine but may not be obvious for some of svelte users.
```typescript
let show: boolean;
$: show = !!data.someKey;
// No errors after that
```

**Comment by dummdidumm** [comment]
This is something we cannot change I think, but it should be documented.

**Comment by jasonlyu123** [comment]
I have [some experiment with implementing]( https://github.com/jasonlyu123/language-tools/tree/reactive-assignment) infer type for reactive assignment. Maybe you can give it a try following [this step](https://github.com/sveltejs/language-tools#development) to see there is any problem.

**Comment by dkzlv** [comment]
@dummdidumm Thanks for your work! Svelte has been the best for me for a long time, but now with this project I just cannot imagine what would make me leave this ecosystem ❤️ 

Back to topic.
Great! To me it is completely ok, but I also think it needs some documentation. I've converted a whole bunch of components to Typescript and encountered some quirks that were difficult to handle (the most difficult one was to find out how I should import interfaces and types in the component, took me a while to figure that out). I would be happy to help with documentation, but I do not see any in the repo. Can you point me to the place?

**Comment by jasonlyu123** [comment]
I also think we should have some typescript documentation. not just like how to type a svelte event handler. There's some svelte specific syntax got transformed to do type check, like what we did in #259, for example. It could be better if we have documentation about it.

**Comment by jasonlyu123** [comment]
@dkzlv most of the docs is in packages/svelte-vscode

**Comment by dkzlv** [comment]
@jasonlyu123 Yeah, I saw it but there's basically no documentation on usage. I'm not sure if I should add it to typescript setup or to the root readme file.

**Comment by jasonlyu123** [comment]
Because it is also related to svelte-check. I personally would prefer it to be in root docs folder in a dedicated file.

**Comment by dummdidumm** [comment]
Wie have the folder `docs` at the root, maybe we should put it in there and also move some preprocessor docs there. Multiple packages rely on those docs so this feels like the right place.

**Comment by jasonlyu123** [comment]
Maybe we could convert some of the examples in official docs to typescript

**Comment by dummdidumm** [comment]
Fixed through #335
